### PR TITLE
Add custom rust toolchain support

### PR DESCRIPTION
**Pull request by fewensa**
- Update follow https://github.com/paritytech/srtool/pull/68

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -216,7 +216,7 @@ If unset, the two envs will have the following default values:
 
 -   `PARACHAIN_PALLET_ID`: `0x01`
 
--   `AUTHORIZE_UPGRADE_PREFIX`: `0x03`
+-   `AUTHORIZE_UPGRADE_PREFIX`: `0x02`
```
Comment (chevdor) [inline README.md:219]: The `README.md` file is generated from the `.adoc` file. Make sure to update the .adoc as well.

Comment (fewensa) [reply README.md:219]: If this https://github.com/chevdor/srtool-actions/pull/22/files#diff-e2d35707e12980263abed184f398892e5cafa7517654b2dc9f71b31f151aa37cR9  it's also updated.

Comment (chevdor) [reply README.md:219]: Indeed, my bad, all good, thank you

**Comment by chevdor** [comment]
## Custom `RUSTC_VERSION`

Picking the `RUSTC_VERSION` is not implemented in `srtool-actions` on purpose.

The idea of fetching the canonical version from `https://raw.githubusercontent.com/${{ inputs.image }}/master/RUSTC_VERSION` is to ensure that all users use the same canonicial version so that any user can check any runtime without having to worry about what srtool to use. It has its short-comings and having the information of the RUSTC_VERSION in the runtime itself is an option I am considering.

At the moment, If you start building your runtime with something else than the canocial version, the checks will fail and the users of your runtime will wonder what the problem is.

Could you explain what your use case is ?

**Comment by fewensa** [comment]
Two reasons.
1. not all like substrate projects use same rust toolchain with substrate/polkadot
2. because of you published docker image tag named semantic style version. but some project use date named version number. https://github.com/darwinia-network/darwinia/blob/3b4dbf996945f1b5237fd5834c0cc38cbc446ceb/rust-toolchain.toml#L2

and srtool build in runtime directory, so can not read root directory `rust-toolchain.toml` file do override rust toolchain (I tested it's not work). so hopefully srtool can set rustc_version

**Comment by chevdor** [comment]
> not all like substrate projects use same rust toolchain with substrate/polkadot

Ok, fair enough. `substrate/polkadot` use the latest stable. it should be a goal to get there. It is possible since recently and you should really aim for that.

> because of you published docker image tag named semantic style version

This is mainly for users to find they way around since currently, they will not have an easy way to know when you built your  runtimes.

> and srtool build in runtime directory, so can not read root directory rust-toolchain.toml file do override rust toolchain (I tested it's not work)

This is blocked on purpose yes

Let me think how we can solve your problem without comprising too much for the users.

**Comment by AurevoirXavier** [comment]
> This is blocked on purpose yes
> 
> Let me think how we can solve your problem without comprising too much for the users.

Currently, I am using a soft link to resolve this issue.

**Comment by AurevoirXavier** [comment]
Hey @chevdor, any feedback?

**Comment by chevdor** [comment]
I am not completely happy with the raw solution as it is really a foot gun.

Nothing against the PR, my main concern is related to users trying to verify runtimes. With this PR in, once the feature is used to build runtimes, those will appear "invalid" unless users can find and use the right version.

I have some ideas how to fix that cleanly but that still requires testing.

**Comment by AurevoirXavier** [comment]
> I am not completely happy with the raw solution as it is really a foot gun.

Sorry, I mean this.

> Currently, I am using a soft link to resolve this issue.

I think maybe we can close this PR.

**Comment by chevdor** [comment]
It would be nice to hear from @fewensa whether your "outlaw" solution 😄 is an acceptable workaround.

**Comment by fewensa** [comment]
I still hope to be implemented by srtool-actions, because this is more general. of course softlink it's works for me

**Pull request closed without merging**